### ISSUE TITLE
Add standalone cross_entropy_loss() function for one-hot targets

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#280]
## Problem

Aprender's `src/loss/` module has standalone loss functions for common losses:

- `mse_loss(&Vector<f32>, &Vector<f32>) -> f32`
- `mae_loss(&Vector<f32>, &Vector<f32>) -> f32`
- `huber_loss(&Vector<f32>, &Vector<f32>, f32) -> f32`

But there is **no standalone cross-entropy function** that takes one-hot targets. The existing options are:

1. `CrossEntropyLoss` struct in `src/nn/loss.rs` — works with `Tensor` objects and class indices, not `Vector<f32>` with one-hot encoding
2. `cross_entropy_with_smoothing()` in `src/regularization/` — takes `target_idx: usize`, not one-hot vector
3. `cross_entropy()` in `src/online/distillation.rs` — uses `f64`, not `f32`

## Proposed Addition

```rust
/// Cross-entropy loss with one-hot targets
///
/// L = -sum(y_true * log(softmax(y_pred)))
pub fn cross_entropy_loss(y_pred: &Vector<f32>, y_true: &Vector<f32>) -> f32 {
    // Numerically stable softmax + CE
}
```

This would follow the same pattern as `mse_loss`/`mae_loss`/`huber_loss` — a simple function taking two `Vector<f32>` references and returning a scalar.

## Context

Entrenar's `CrossEntropyLoss` computes softmax + CE inline because there's no matching aprender standalone function to delegate to. All other loss forward computations (MSE, L1, Huber) now delegate to aprender.

Cross-ref: entrenar `docs/specifications/simplify-entrenar.md` Section 2.3
